File: src/main/java/org/bakkes/game/model/AModel.java
package org.bakkes.game.model;


public class AModel implements IModel{

	private String name;

	public AModel(){
		name = "anonymos";
	}


	@Override
	public String getName() {
		return name;
	}

	@Override
	public void setName(final String name) {
		assert name != null;
		this.name = name;
	}
	@Override
	public String toString(){
		return this.getClass().getSimpleName() + " " + this.getName() + " ";
	}
}

File: src/main/scala/org/bakkes/game/model/AModel.scala
package org.bakkes.game.model

abstract class AModel extends IModel {
	var name: String = ""
	override def toString: String = {
		this.getClass.getSimpleName + " "  + this.getName + " "
	}

	override def getName = name
	override def setName(to:String) = name = to
}